types/relay-runtime/index.d.ts

```diff
@@ -0,0 +1,1130 @@
+// Type definitions for relay-runtime 1.3
+// Project: https://github.com/facebook/relay
+// Definitions by: Matt Martin <https://github.com/voxmatt>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+// note: namespace is used so react-relay can access and inter-op
+export namespace RelayCommonTypes {
```

Comment:
Why does this need to be in a namespace? You could directly export types from the module and import them as in `import { RelayConcreteNode } from "relay-runtime";`.

Reply:
Addressed in https://github.com/voxmatt/DefinitelyTyped/pull/2

Reply:
I merged @alloy's PR  because it's just awesome to have the help and it was a big improvement. But I did prefer my pattern of only exporting the exports of the source package. The namespace enabled access to the underlying types without polluting the package's exports. Now, `import { ... } from 'react-relay'` exposes all sorts of cruft that makes it harder to understand the source package.

Reply:
I agree with this, but then we should just model the various modules (files) the same way they are in the distribution library. E.g. rather than having a `RelayNetworkTypes` section inside `types/relay-runtime/index.d.ts`, they should go into  `types/relay-runtime/lib/RelayNetworkTypes.d.ts`.

I haven’t taken a stab at this yet, because I wanted to reach consensus first and have limited time. If you agree with this approach then we should do so before merging, otherwise we’d introduce breaking changes next time.

Reply:
It's pretty standard on DefinitelyTyped to bundle types for a library's interfaces along with types for its functions -- I would think it would be annoying to have to import from a different file just to get access to interfaces.

Reply:
Totally fair. I'm in agreement with you two and, @alloy, thanks again for all the work getting this over the finish line. I took a weekend getting this in place and then have been overwhelmed with work ever since.

Reply:
Ace 👌 I had the exact same situation, so luckily we could collab. High fives all around!